frontend/src/pages/AccountReceive.vue

```diff
@@ -178,40 +179,92 @@ function useScan() {
     const overrides = { startBlock: startBlockLocal.value, endBlock: endBlockLocal.value };
-    let allAnnouncements: AnnouncementDetail[] = [];
-    try {
-      if (!signer.value) throw new Error('signer is undefined');
-      if (!userWalletAddress.value) throw new Error('userWalletAddress is undefined');
-      // When private key is provided in advanced mode, we fetch all announcements
-      if (advancedMode.value && scanPrivateKey.value)
-        allAnnouncements = await umbra.value.fetchAllAnnouncements(overrides);
-      else
-        allAnnouncements = await umbra.value.fetchSomeAnnouncements(signer.value, userWalletAddress.value, overrides);
-    } catch (e) {
-      scanStatus.value = 'waiting'; // reset to the default state because we were unable to fetch announcements
-      throw e;
-    }
 
     // Scan for funds
-    scanStatus.value = 'scanning';
     const spendingPubKey = chooseKey(spendingKeyPair.value?.publicKeyHex);
     const viewingPrivKey = chooseKey(viewingKeyPair.value?.privateKeyHex);
 
-    scanPercentage.value = 0;
-    filterUserAnnouncements(
-      spendingPubKey,
-      viewingPrivKey,
-      allAnnouncements,
-      (percent) => {
-        scanPercentage.value = Math.floor(percent);
-      },
-      (filteredAnnouncements) => {
-        userAnnouncements.value = filteredAnnouncements.sort(function (a, b) {
-          return parseInt(a.timestamp) - parseInt(b.timestamp);
-        });
+    // Wrapper for `filterUserAnnouncements` so we can await for the first batch of web workers to finish before
+    // creating new workers.
+    const filterUserAnnouncementsAsync = (
+      spendingPublicKey: string,
+      viewingPrivateKey: string,
+      announcements: AnnouncementDetail[]
+    ) => {
+      scanStatus.value = 'scanning';
+      return new Promise<void>((resolve) => {
+        filterUserAnnouncements(
+          spendingPublicKey,
+          viewingPrivateKey,
+          announcements,
+          (percent) => {
+            scanPercentage.value = Math.floor(percent);
+          },
+          (filteredAnnouncements) => {
+            userAnnouncements.value = [...userAnnouncements.value, ...filteredAnnouncements].sort(function (a, b) {
+              return parseInt(a.timestamp) - parseInt(b.timestamp);
+            });
+            if (filterUserAnnouncements.length) tableKey.value += 1;
+            scanStatus.value = 'complete';
+            scanPercentage.value = 0;
+            resolve();
+          }
+        );
+      });
+    };
+
+    try {
+      if (!signer.value) throw new Error('signer is undefined');
+      if (!userWalletAddress.value) throw new Error('userWalletAddress is undefined');
+
+      let announcementsCount = 0; // Track the count of announcements
+      let announcementsQueue: AnnouncementDetail[] = []; // Announcements to be filtered
+      let firstScanPromise = Promise.resolve();
+      // When in advanced mode, we fetch all announcements
+      if (advancedMode.value) {
```

Comment:
Why do we always fetch all announcements in advanced mode? I think we can remove this `if` block and use the same logic regardless of advanced mode

Reply:
Latest behavior is if `advancedMode` is on and `scanPrivateKey` is provided, we fetch for all announcements.

Reply:
Why do we not respect user-provided block ranges if a private key is provided in advanced mode?

Reply:
Actually we do. We pass in `overrides` to `fetchAllAnnouncements`, which will include user specified block ranges.

Reply:
Ok, I see the overrides being passed. So the comment in https://github.com/ScopeLift/umbra-protocol/pull/558/files#diff-27047766a3ad711f41f2a80bc8036e498999c7963b3196e7b9e4f27ae3e5804eR256-R258 needs to be updated then?

Reply:
Updated here : https://github.com/ScopeLift/umbra-protocol/pull/558/commits/599d33aead443ef144250df11c5d6f8714004b48